lib/inertia_rails/middleware.rb

```diff
@@ -71,17 +74,15 @@ def inertia_request?
       end
 
       def version_stale?
-        sent_version != saved_version
+        coerce_version(client_version) != coerce_version(server_version)
       end
 
-      def sent_version
-        return nil if inertia_version.nil?
-
-        InertiaRails.version.is_a?(Numeric) ? inertia_version.to_f : inertia_version
+      def server_version
+        controller.send(:inertia_configuration).version
```

Comment:
Controller might be nil (i.e. in case of 404 error), I think this is a valid fix:

```suggestion
        config = controller ? controller.send(:inertia_configuration) : ::InertiaRails.configuration
        config.version
```

Reply:
Yep, I confirmed this locally. You'd need to make an Inertia request and also use the Rails default 404 handling, but it does indeed throw an error.

I'm gonna have it just return nil instead of trying to fall back to a global configuration; since we can have multiple versions configured now I think we shouldn't try to guess at the version here.

```suggestion
        controller&.send(:inertia_configuration)&.version
```